dipy/reconst/tests/test_mapmri.py

```diff
@@ -60,45 +188,463 @@ def test_mapmri_metrics():
 
     assert_almost_equal(integral, 1.0, 3)
 
-    # compare the shore pdf with the ground truth multi_tensor pdf
+    # test if numerical integral of odf is equal to one
+    odf = mapfit.odf(sphere, s=0)
+    odf_sum = odf.sum() / sphere.vertices.shape[0] * (4 * np.pi)
+    assert_almost_equal(odf_sum, 1.0, 2)
 
-    sphere = get_sphere('symmetric724')
-    v = sphere.vertices
-    radius = 10e-3
-    r_points = v * radius
+    # do the same for isotropic implementation
+    radius_max = 0.04  # 40 microns
+    gridsize = 17
+    r_points = mapmri.create_rspace(gridsize, radius_max)
+    mapm = MapmriModel(gtab, radial_order=radial_order,
+                       laplacian_weighting=0.02,
+                       anisotropic_scaling=False)
+    mapfit = mapm.fit(S)
+    pdf = mapfit.pdf(r_points)
+    pdf[r_points[:, 2] == 0.] /= 2  # for antipodal symmetry on z-plane
+
+    point_volume = (radius_max / (gridsize // 2)) ** 3
+    integral = pdf.sum() * point_volume * 2
+    assert_almost_equal(integral, 1.0, 3)
+
+    odf = mapfit.odf(sphere, s=0)
+    odf_sum = odf.sum() / sphere.vertices.shape[0] * (4 * np.pi)
+    assert_almost_equal(odf_sum, 1.0, 2)
+
+
+def test_mapmri_compare_fitted_pdf_with_multi_tensor(radial_order=6):
+    gtab = get_gtab_taiwan_dsi()
+    l1, l2, l3 = [0.0015, 0.0003, 0.0003]
+    S, _ = generate_signal_crossing(gtab, l1, l2, l3)
+
+    radius_max = 0.02  # 40 microns
+    gridsize = 10
+    r_points = mapmri.create_rspace(gridsize, radius_max)
+
+    # test MAPMRI fitting
+    mapm = MapmriModel(gtab, radial_order=radial_order,
+                       laplacian_weighting=0.0001)
+    mapfit = mapm.fit(S)
+
+    # compare the mapmri pdf with the ground truth multi_tensor pdf
+
+    mevals = np.array(([l1, l2, l3],
+                       [l1, l2, l3]))
+    angl = [(0, 0), (60, 0)]
     pdf_mt = multi_tensor_pdf(r_points, mevals=mevals,
                               angles=angl, fractions=[50, 50])
-
-    pdf_map = mapmri_EAP(r_points, radial_order, c_map, mu, R)
+    pdf_map = mapfit.pdf(r_points)
 
     nmse_pdf = np.sqrt(np.sum((pdf_mt - pdf_map) ** 2)) / (pdf_mt.sum())
     assert_almost_equal(nmse_pdf, 0.0, 2)
 
-    # test MAPMRI metrics
+
+def test_mapmri_metrics_anisotropic(radial_order=6):
+    gtab = get_gtab_taiwan_dsi()
+    l1, l2, l3 = [0.0015, 0.0003, 0.0003]
+    S, _ = generate_signal_crossing(gtab, l1, l2, l3, angle2=0)
+
+    # test MAPMRI q-space indices
+
+    mapm = MapmriModel(gtab, radial_order=radial_order,
```

Comment:
General question: would it make sense to test different radial orders? I see that you set this in the function signature in many places as a default to 6, but I wonder whether it would make sense to iterate (at least in one of the tests) over several plausible values.